canonical/ubuntu.com
# WD-14515 - Update package.json major dependencies

This PR is to fix part of https://github.com/canonical/ubuntu.com/pull/12389

## Done

- Update package.json dependencies
- Migrate code using date-fns to newer version
- Fix incorrect types of `start_date` and `end_date` in `UserSubscription` type

## QA

- Go to the demo website.
- Check all pages are working as expected (especially `/pro/dashboard`).

## Issue / Card

Fixes [WD-14515](https://warthogs.atlassian.net/browse/WD-14515)

## Help

[QA steps](https://discourse.canonical.com/t/qa-steps/152) - [Commit guidelines](https://discourse.canonical.com/t/commit-guidelines/148)


[WD-14515]: https://warthogs.atlassian.net/browse/WD-14515?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
